marimbabot_vision/src/vision_node.py

```diff
@@ -1,25 +1,85 @@
 #!/usr/bin/env python
+
+import numpy as np
 import rospy
+import torch
+from PIL import Image as PILImage
+from sensor_msgs.msg import Image as ROSImage
 from std_msgs.msg import String
-from sensor_msgs import Image
+from transformers import (DonutProcessor, VisionEncoderDecoderConfig,
+                          VisionEncoderDecoderModel)
 
-def callbackImage(data):    # TODO
-    pass
-#  rospy.loginfo()
-    
-def listener():
+def detect_notes(img_array: np.array, pre_processor: DonutProcessor, model: VisionEncoderDecoderModel):
+    # Load image
+    image = PILImage.open(img_array).convert('RGB')
+
```

Comment:
We might need to rescale the image at some point, because the camera image will be much more high res than the model input. It could be that the donut preprocessor is already doing that, but I would crosscheck this.

Reply:
Yes, it should rescale it. From the docs I got that there is a parameter "do_rescale".

DonutProcessor is instance of DonutImageProcessor. There the default value for the parameter is set to True.

https://huggingface.co/docs/transformers/model_doc/donut#transformers.DonutImageProcessor

Reply:
Okay nice 👍